app/src/main/java/com/andrew/tasky/agenda/presentation/adapters/PhotoItemAdapter.kt

```diff
@@ -1,26 +1,31 @@
 package com.andrew.tasky.agenda.presentation.adapters
 
+import android.content.Context
 import android.net.Uri
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
+import com.andrew.tasky.agenda.domain.models.EventPhoto
 import com.andrew.tasky.agenda.domain.models.Photo
+import com.andrew.tasky.agenda.presentation.screens.event_detail.EventDetailFragment
 import com.andrew.tasky.databinding.ItemPhotoAdapterCardBinding
+import com.bumptech.glide.Glide
 
 class PhotoItemAdapter(
+    val context: Context,
     private val onPhotoClick: (Int) -> Unit,
     private val onAddPhotoClick: () -> Unit,
     private val userIsAttendee: Boolean
-) : ListAdapter<Photo, PhotoItemAdapter.PhotoItemViewHolder>(Companion) {
+) : ListAdapter<EventPhoto, PhotoItemAdapter.PhotoItemViewHolder>(Companion) {
 
-    companion object : DiffUtil.ItemCallback<Photo>() {
-        override fun areItemsTheSame(oldItem: Photo, newItem: Photo): Boolean {
+    companion object : DiffUtil.ItemCallback<EventPhoto>() {
+        override fun areItemsTheSame(oldItem: EventPhoto, newItem: EventPhoto): Boolean {
             return oldItem == newItem
         }
 
-        override fun areContentsTheSame(oldItem: Photo, newItem: Photo): Boolean {
+        override fun areContentsTheSame(oldItem: EventPhoto, newItem: EventPhoto): Boolean {
             return oldItem == newItem
         }
     }
```

Comment:
Careful here, since your `EventPhoto.Local` is not a data class, therefore it does not have an actual equals function to compare its content. So `localPhoto1 == localPhoto2` will only return true if it's the same reference, but not necessarily if it's the same Uri (which you also wouldn't want since a user could add the same photo twice).

I'd recommend also using keys for local photos which you can randomly generate when you add one (using `UUID.randomUUID().toString()`). That way you can better compare if 2 local photos are really the same or not.